RSyntaxTextArea/src/main/java/org/fife/ui/rsyntaxtextarea/modes/IniTokenMaker.flex

```
@@ -72,151 +72,108 @@ import org.fife.ui.rsyntaxtextarea.*;
-			return yylex();
-		} catch (IOException ioe) {
-			ioe.printStackTrace();
-			return new TokenImpl();
-		}
-
-	}
-
-
-	/**
-	 * Refills the input buffer.
-	 *
-	 * @return      <code>true</code> if EOF was reached, otherwise
-	 *              <code>false</code>.
-	 */
-	private boolean zzRefill() {
-		return zzCurrentPos>=s.offset+s.count;
-	}
-
-
-	/**
-	 * Resets the scanner to read from a new input stream.
-	 * Does not close the old reader.
-	 *
-	 * All internal variables are reset, the old input stream 
-	 * <b>cannot</b> be reused (internal buffer is discarded and lost).
-	 * Lexical state is set to <tt>YY_INITIAL</tt>.
-	 *
-	 * @param reader   the new input stream 
-	 */
-	public final void yyreset(Reader reader) {
-		// 's' has been updated.
-		zzBuffer = s.array;
-		/*
-		 * We replaced the line below with the two below it because zzRefill
-		 * no longer "refills" the buffer (since the way we do it, it's always
-		 * "full" the first time through, since it points to the segment's
-		 * array).  So, we assign zzEndRead here.
-		 */
-		//zzStartRead = zzEndRead = s.offset;
-		zzStartRead = s.offset;
-		zzEndRead = zzStartRead + s.count - 1;
-		zzCurrentPos = zzMarkedPos = zzPushbackPos = s.offset;
-		zzLexicalState = YYINITIAL;
-		zzReader = reader;
-		zzAtBOL  = true;
-		zzAtEOF  = false;
-	}
-
-
+	void addToken(int tokenType) {
+        super.addToken(s.array, s.offset + zzStartRead, s.offset + zzMarkedPos - 1, tokenType, start + zzStartRead);
+    }
+
+    @Override
+    public String[] getLineCommentStartAndEnd(int languageIndex) {
+        return new String[]{"#", null};
+    }
+
+    @Override
+    public Token getTokenList(Segment text, int initialTokenType, int startOffset) {
+        resetTokenList();
+
+        try {
+            super.s = text;
+            super.start = startOffset;
+            reset(text, 0, text.count, 0);
+            yylex();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

This will crash some apps if there is a bug in the highlighting. While RSTA's current behavior of silently returning not text isn't great, I'd rather keep the current behavior consistent across all `TokenMaker`s.

---

This and the previous issue have been fixed: the semicolon is restored as comment marker and no exception is thrown during unexpected lexical errors.